pilot/pkg/model/gateway.go

```diff
@@ -253,6 +265,23 @@ func MergeGateways(gateways []gatewayWithInstances, proxy *Proxy) *MergedGateway
 						} else {
 							mergedServers[serverPort].Servers = append(mergedServers[serverPort].Servers, s)
 						}
+
+						// We have TLS settings defined and we have already taken care of unique route names
+						// if it is HTTPS. So we can construct a QUIC server on the same port. It is okay as
+						// QUIC listens on UDP port, not TCP
+						if features.EnableQUICListeners && gateway.IsEligibleForHTTP3Upgrade(s) &&
```

Comment:
nit: IsEligibleForHTTP3Upgrade already checks EnableQUICListeners, no need to replace

Reply:
Yeah. I see it is redundant. Guess it is fine for readability?